[Build Speed] Make GraphicsLayer.h less expensive

jernoble opened the pull request:
#### 0635672c86b4a255ce06cea8223b248703ee6151
<pre>
[Build Speed] Make GraphicsLayer.h less expensive
<a href="https://rdar.apple.com/161829723">rdar://161829723</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=300033">https://bugs.webkit.org/show_bug.cgi?id=300033</a>

Reviewed by Ryosuke Niwa.

Prior to this patch, GraphicsLayer.h was the 3rd most expensive header used in
WebCore; it is included 268 times in a WebCore Unified build, and on this
machine, each include of that file took the compiler on avegare 1.3s to parse,
for a total of 5.9m CPU minutes of time spent parsing this header.

The primary cause of GraphicsLayer.h being included in a compliation unit was
that it was included by GraphicsLayerContentsDisplayDelegate.h, which only
needed a forward declaration of an enum defined within the GraphicsLayer class
itself. This enum–along with every other enum defined within GraphicsLayer–was
moved into a new GraphicsLayerEnums.h header, and those enums were pulled back
into GraphicsLayer via a using statement. I then looked at every header which
included GraphicsLayer.h and: converted every use of GraphicsLayer enums to use
the new enums defined in GraphicsLayerEnums.h, added a forward declaration for
GraphicsLayer, and removed the GraphicsLayer.h header where possible.

After this patch, GraphicsLayer.h is now the 67th most expensive header used in
WebCore; it is included 37 times, for a total of 49s CPU seconds of time spent
parsing this header, a reduction of 5.1m.

Canonical link: <a href="https://commits.webkit.org/301010@main">https://commits.webkit.org/301010@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e07a65c3442a184963a548a803d375133132ef8c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124668 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44340 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35074 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131509 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76597 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a7a19f94-34c2-47b3-8560-ebbe2f055847) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126545 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45044 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52910 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94851 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62907 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/29948f9f-6bf5-4f4a-9700-cfef2c082c9c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127622 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35919 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111492 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75421 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ba76ce2b-a06b-422b-8d60-1840ddbc5b82) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34849 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29647 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74989 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105676 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29878 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134175 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51516 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39332 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103324 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51928 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107713 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103099 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26244 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48470 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26735 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48472 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51390 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57187 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50784 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54139 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52478 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c6bb2fa189cb34c4216fcbba9085f06ea6e8590f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51696

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b514e50683c92bcd1db1450eb956d577dadbca04. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51696

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e3b5a8cf6499fee6846583442669cdbfc9d466fa. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51696

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e07a65c3442a184963a548a803d375133132ef8c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51696

jernoble committed:
[Build Speed] Make GraphicsLayer.h less expensive
rdar://161829723
https://bugs.webkit.org/show_bug.cgi?id=300033

Reviewed by Ryosuke Niwa.

Prior to this patch, GraphicsLayer.h was the 3rd most expensive header used in
WebCore; it is included 268 times in a WebCore Unified build, and on this
machine, each include of that file took the compiler on avegare 1.3s to parse,
for a total of 5.9m CPU minutes of time spent parsing this header.

The primary cause of GraphicsLayer.h being included in a compliation unit was
that it was included by GraphicsLayerContentsDisplayDelegate.h, which only
needed a forward declaration of an enum defined within the GraphicsLayer class
itself. This enum–along with every other enum defined within GraphicsLayer–was
moved into a new GraphicsLayerEnums.h header, and those enums were pulled back
into GraphicsLayer via a using statement. I then looked at every header which
included GraphicsLayer.h and: converted every use of GraphicsLayer enums to use
the new enums defined in GraphicsLayerEnums.h, added a forward declaration for
GraphicsLayer, and removed the GraphicsLayer.h header where possible.

After this patch, GraphicsLayer.h is now the 67th most expensive header used in
WebCore; it is included 37 times, for a total of 49s CPU seconds of time spent
parsing this header, a reduction of 5.1m.

Canonical link: https://commits.webkit.org/301010@main
